.github/workflows/ci.yml

```diff
@@ -3,35 +3,50 @@ name: CI
 on:
   pull_request:
   push:
-    branches: [ main ]
+    branches: [main]
+
+permissions:
```

Comment:
This is recommended practice for GitHub Actions (GHA) for controlling the permissions allowed to the `GITHUB_TOKEN`: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token